component/status/status.go

```diff
@@ -0,0 +1,131 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package status
```

Comment:
Reading this package, makes me think that we want this in `component`, because all of the structs start with "component".

Reply:
I would rename `component.StatusEvent` the main struct of this package and get rid of it.

Reply:
That was my initial approach, but it pollutes the component package with many new public symbols. I think it is a bit better this way to isolate status-related stuff in a separate sub-package. No strong opinion, I can merge if we think that's better.